SCClassLibrary/Common/Core/Kernel.sc

```diff
@@ -549,6 +549,56 @@ FunctionDef {
 		^().putPairs(this.keyValuePairsFromArgs)
 	}
 
+
+	prMakePerformableEnvir {|argsArray=([]), keywordArgsEvent=(()), dropFirstN=0|
+		var argNamesNoThis = argNames.drop(dropFirstN);
+		var setOfAddedNames = Set();
+		var out = ();
+		// skips this
+		argsArray.do{|ar, i|
+			out[argNamesNoThis[i]] = ar;
+			setOfAddedNames.add(argNamesNoThis[i]);
+		};
+
+		keywordArgsEvent.keysValuesDo{|k, v|
+			var index;
+			if(setOfAddedNames.includes(k)) {
+				Error("Argument with name % already added.".format(k)).throw
+			};
+
+			if(argNamesNoThis.includes(k).not) {
+				Error("Argument '%' not found in method '%'".format(k, this)).throw
+			};
+
+			out[k] = v;
+			setOfAddedNames.add(k);
+		};
+		^out
+	}
+
+	prMakePerformableArray {|argsArray=([]), keywordArgsEvent=(()), dropFirstN=0|
+		var ev = this.makePerformableEnvir(argsArray, keywordArgsEvent, dropFirstN);
+		var protoDrop = prototypeFrame.drop(dropFirstN);
```

Comment:
you are not using `protoDrop` afterwards.

But if, you could use the suggestion above and get rid of `dropFirstN`